[ci] Update start-core.sh to create default wallet for bitcoind 0.21.0

notmandatory opened the pull request:
### Description

The new bitcoin core release `0.21.0` does not create a default wallet. We need to create a default wallet ourselves in the `start-core.sh` script so that electrum (and future) integration tests will work.

To support this change I also created a new docker image `bitcoindevkit/esplora:0.2.0` which includes `bitcoind` `0.21.0` and enables compact block filters, see https://github.com/bitcoindevkit/bitcoin-regtest-box/pull/3.

Fixes #267 

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [x] I'm linking the issue being fixed by this PR

notmandatory committed:
[ci] Update start-core.sh to create default wallet for bitcoind 0.21.0

notmandatory commented:
Creating a new wallet with our current bitcoin `0.20.1` doesn't break anything so we can merge this any time prior to upgrading our `bitcoin-regtest-box` containers to `0.21.0` https://github.com/bitcoindevkit/bitcoin-regtest-box/pull/3.

shesek commented:
> Creating a new wallet with our current bitcoin `0.20.1 `doesn't break anything

With 0.20.1 that already has a default wallet, wouldn't creating the second `bdk-test` wallet make it necessary to specify the `rpcwallet` for all wallet-related rpc calls?

For example, it looks like the `getnewaddress` call a couple lines below `createwallet` should be broken when testing against 0.20.

notmandatory commented:
> With 0.20.1 that already has a default wallet, wouldn't creating the second `bdk-test` wallet make it necessary to specify the `rpcwallet` for all wallet-related rpc calls?

Yes, you are correct, I did some more testing and it looks like with `0.20.1` with both "" and "bdk-test" wallets we do need to specify the rpcwallet name. But with `0.21.0` with just one wallet (ie. `bdk-test`) the rpc commands default to that wallet. I'll move this back to Draft until https://github.com/bitcoindevkit/bitcoin-regtest-box/pull/3 is merged.

shesek commented:
I would consider changing the tests to explicitly specify the wallet, so the tests would also work against older Bitcoin Core versions.

notmandatory committed:
[ci] Rename MAGICAL_ env vars to BDK_, for tests use wallet name in RPC calls

notmandatory commented:
> I would consider changing the tests to explicitly specify the wallet, so the tests would also work against older Bitcoin Core versions.

Fixed. I added explicit wallet selection in `ci/start-core.sh` and in `testutils::TestClient`.

afilini commented on Cargo.toml:
```diff
@@ -59,8 +59,8 @@ test-electrum = ["electrum"]
 [dev-dependencies]
-bdk-testutils = "^0.3"
-bdk-testutils-macros = "^0.3"
+bdk-testutils = { path = "./testutils"}
```
I'm thinking that maybe we should always use local dependencies in our development branches and only replace them with version number right before the release. It would need to change the way we do releases though, because we would have to merge the release branch before we do the bump. Something like:

```
master -------------------------------------------- bump dev ------ ...
           \                                  /
release     --- bump rc.1 --- fix --- fix --- ---- publish deps --- bump release version
```

Don't know if this would be better, just thinking out loud

notmandatory replied:
Yes I agree,  I ran into the same issue when testing locally. We should also add a step to the DEVELOPMENT_CYCLE.md so we don't forget.. but I'll do that in a separate PR.

afilini commented:
Needs a rebase I think and then we can merge this

notmandatory commented:
> Needs a rebase I think and then we can merge this

I think the problem is that the `Audit` and `Code Coverage` workflows were marked as required, but they only run when code is actually pushed to master, not on PRs. I might have messed it up when I changed the required checks for rust `1.50.0`. Github checks all pass now so I'll merge it.
